base/process.jl

```diff
@@ -714,6 +714,7 @@ ProcessExitedError(proc::Process) = ProcessExitedError([proc])
 
 function show(io::IO, err::ProcessExitedError)
     if length(err.procs) == 1
+        @inbounds proc = err.procs[1]
```

Comment:
This `@inbounds` seems unnecessary.

Reply:
Because this is not in a situation where it will improve performance (nor where performance matter) and even though the length is explicitly checked above it adds extra code and if the check, by mistake, later is removed or changed, the inbounds will give silent corruption instead of an error.

Reply:
fair enough, fixed